addons/binding/org.openhab.binding.hs110/src/main/java/org/openhab/binding/hs110/handler/HS110Handler.java

```diff
@@ -0,0 +1,225 @@
+                } catch (IOException e) {
+                    logger.error("Failed sending command {} to {}", command.toFullString(),
+                            thing.getUID().getAsString(), e);
+                }
+            }
+        }
+
+        if (command instanceof RefreshType) {
+            boolean success = updateData();
+            if (success) {
+                switch (channelUID.getId()) {
+                    case CHANNEL_WATTAGE:
+                        updateState(channelUID, getWattage());
+                        break;
+                    case CHANNEL_TOTAL:
+                        updateState(channelUID, getTotal());
+                        break;
+                    case CHANNEL_SWITCH:
+                        updateState(channelUID, getState());
+                        break;
+                    case CHANNEL_SYSINFO:
+                        updateState(channelUID, getSysinfo());
+                        break;
+                    default:
+                        logger.debug("Command received for an unknown channel: {}", channelUID.getId());
+                        break;
+                }
+            }
+        } else {
+            logger.debug("Command {} is not supported for channel: {}", command, channelUID.getId());
+        }
+
+    }
+
+    @Override
+    public void dispose() {
+        refreshJob.cancel(true);
+    }
+
+    @Override
+    public void initialize() {
+        Configuration config = getThing().getConfiguration();
+
+        String ip = (String) config.get(CONFIG_IP);
+        logger.info("Initializing plug on ip {}", ip);
+        plug = new HS110(ip);
+
+        if (config.containsKey(CONFIG_REFRESH)) {
+            refresh = (BigDecimal) config.get(CONFIG_REFRESH);
+        } else {
+            refresh = new BigDecimal(REFRESH_DEFAULT);
+        }
+
+        startAutomaticRefresh();
+
+        updateStatus(ThingStatus.ONLINE);
+
+    }
+
+    private void startAutomaticRefresh() {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                logger.debug("Refreshing {}", thing.getUID().getAsString());
+                try {
+                    boolean success = updateData();
+                    if (success) {
+                        updateState(new ChannelUID(getThing().getUID(), CHANNEL_SWITCH), getState());
+                        updateState(new ChannelUID(getThing().getUID(), CHANNEL_WATTAGE), getWattage());
+                        updateState(new ChannelUID(getThing().getUID(), CHANNEL_TOTAL), getTotal());
+                        updateState(new ChannelUID(getThing().getUID(), CHANNEL_SYSINFO), getSysinfo());
+                    }
+                } catch (Exception e) {
+                    logger.debug("Exception occurred during execution", e);
+                }
+            }
+        };
+
+        refreshJob = scheduler.scheduleAtFixedRate(runnable, 0, refresh.intValue(), TimeUnit.SECONDS);
```

Comment:
Please use [scheduleWithFixedDelay](http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleWithFixedDelay(java.lang.Runnable,%20long,%20long,%20java.util.concurrent.TimeUnit)) which is preferred over scheduleAtFixedRate, see [Coding Guidelines D.3](http://docs.openhab.org/developers/development/guidelines.html#d-runtime-behavior)